### ISSUE TITLE
[Epic] PostgreSQL in service node: Further optimisations

### ISSUE THREAD

**Issue by fryorcraken** [waku-org/pm#84]
**Epic label**: `E:PostgreSQL`

# Summary

Further optimisation of PosgreSQL usage for Waku archive in nwaku.
Specifically to cover the case of Status Communities and 10k users in a given Community/shard/store.

# Acceptance Criteria

- Various optimization/performance fix have been implementing.
- Performance analysis done and follow-up items are implemented.

Note that while go-waku also implements PostgresSQL, the focus is on nwaku as a service node.
Providing store service from Status Desktop app is not part of the scope of this epic.
This will be possible/interesting once #57 is done.

# Tasks

- [x] nwaku: https://github.com/waku-org/nwaku/issues/1894
- [x] nwaku: https://github.com/waku-org/nwaku/issues/1893
- [x] nwaku: https://github.com/waku-org/nwaku/issues/1888
- [ ] nwaku: https://github.com/waku-org/nwaku/issues/1885
- [x] nwaku: https://github.com/waku-org/nwaku/issues/1842
- [ ] nwaku: https://github.com/waku-org/nwaku/issues/1841
- [x] nwaku: https://github.com/waku-org/nwaku/issues/1840
- [x] nwaku: https://github.com/waku-org/nwaku/issues/1604

# RAID (Risks, Assumptions, Issues and Dependencies)
